src/org/terraform/utils/GenUtils.java

```diff
@@ -91,15 +91,8 @@ public static boolean chance(int chance, int outOf) {
         
         for (int x = gridX; x < gridX + 16; x++) {
             for (int z = gridZ; z < gridZ + 16; z++) {
-                int height = HeightMap.getHeight(tw, x, z);//GenUtils.getTrueHighestBlock(data, x, z);
-                for (BiomeBank bank : BiomeBank.VALUES) {
-                    BiomeBank currentBiome = tw.getBiomeBank(x, height, z);//BiomeBank.calculateBiome(tw,tw.getTemperature(x, z), height);
-
-                    if (bank == currentBiome) {
-                        if (!banks.contains(bank)) banks.add(bank);
-                        break;
-                    }
-                }
+                BiomeBank bank = tw.getBiomeBank(x, HeightMap.getHeight(tw, x, z), z);
+                if (!banks.contains(bank)) banks.add(bank);

```

Comment:
This is more optimized right? Or is there something I don't see in the code logic